optimism/src/keccak/witness.rs

```diff
@@ -72,67 +81,21 @@ impl<F: Field> KeccakInterpreter<F> for Env<F> {
     // LOOKUPS OPERATIONS //
     ////////////////////////
 
-    fn add_lookup(&mut self, _lookup: Lookup<Self::Variable>) {
+    fn add_lookup(&mut self, lookup: Lookup<Self::Variable>) {
         // Keep track of multiplicities for fixed lookups
-        todo!()
-    }
-
-    fn lookup_syscall_preimage(&mut self) {
-        // No-op for witness
-    }
-
-    fn lookup_syscall_hash(&mut self) {
-        // No-op for witness
-    }
-
-    fn lookup_steps(&mut self) {
-        // No-op for witness
-    }
-
-    fn lookup_rc16(&mut self, flag: Self::Variable, _value: Self::Variable) {
-        if flag == Self::one() {
-            // TODO: keep track of multiplicity of range check 16 entry
-            // TODO: check that [value] is in the RangeCheck16Lookup
-        }
-    }
-
-    fn lookup_reset(
-        &mut self,
-        flag: Self::Variable,
-        _dense: Self::Variable,
-        _sparse: Self::Variable,
-    ) {
-        if flag == Self::one() {
-            // TODO: keep track of multiplicity of range check 16 entry
-            // TODO: check that [dense, sparse] is in the ResetLookup
-        }
-    }
-
-    fn lookup_sparse(&mut self, flag: Self::Variable, _value: Self::Variable) {
-        if flag == Self::one() {
-            // TODO: keep track of multiplicity of range check 16 entry
-            // TODO: check that [value] is in the SparseLookup
-        }
-    }
-
-    fn lookup_byte(&mut self, flag: Self::Variable, _value: Self::Variable) {
-        if flag == Self::one() {
-            // TODO: keep track of multiplicity of range check 16 entry
-            // TODO: check that [value] is in the ByteLookup
-        }
-    }
-
-    fn lookup_pad(&mut self, flag: Self::Variable, _value: Vec<Self::Variable>) {
-        if flag == Self::one() {
-            // TODO: keep track of multiplicity of range check 16 entry
-            // TODO: check that [value] is in the PadLookup
-        }
-    }
-
-    fn lookup_round_constants(&mut self, flag: Self::Variable, _value: Vec<Self::Variable>) {
-        if flag == Self::one() {
-            // TODO: keep track of multiplicity of range check 16 entry
-            // TODO: check that [value] is in the RoundConstantsLookup
+        match lookup.table_id {
+            RangeCheck16Lookup | SparseLookup | ResetLookup | RoundConstantsLookup | PadLookup
+            | ByteLookup => {
+                if lookup.magnitude == Self::one() {
+                    // Check that the lookup value is in the table
+                    if let Some(idx) = LookupTable::in_table(lookup.table_id, lookup.value) {
+                        self.multiplicities[lookup.table_id as usize][idx] += 1;
+                    } else {
+                        self.errors.push(Error::Lookup(lookup.table_id));
+                    }
+                }
+            }
+            _ => (),
```

Comment:
(it is a personal taste, but it has been shown to be useful in the past)
I would prefer to list all the other possible variants. This way, if one day we add a variant to the type, the compiler will force us to handle the case manually, and we will have to give a thought on the value to return.